apps/workflows/jobs/spend-management.ts

```diff
@@ -0,0 +1,269 @@
+      const usedVerifications =
+        verifications.find((usage) => usage.workspaceId === workspace.id)?.verifications || 0;
+
+      let currentPrice = 0;
+      if (workspace.subscriptions?.plan) {
+        const cost = parseFloat(workspace.subscriptions.plan.cents);
+        currentPrice += cost;
+      }
+      if (workspace.subscriptions?.support) {
+        const cost = parseFloat(workspace.subscriptions.support.cents);
+        currentPrice += cost;
+      }
+      if (workspace.subscriptions?.activeKeys) {
+        const cost = calculateTieredPrices(
+          workspace.subscriptions.activeKeys.tiers,
+          usedActiveKeys,
+        );
+        // TODO: What to do on error? Skip billing? Assume zero?
+        if (!cost.err) {
+          currentPrice += cost.val.totalCentsEstimate;
+        }
+      }
+      if (workspace.subscriptions?.verifications) {
+        const cost = calculateTieredPrices(
+          workspace.subscriptions.verifications.tiers,
+          usedVerifications,
+        );
+        // TODO: What to do on error? Skip billing? Assume zero?
+        if (!cost.err) {
+          currentPrice += cost.val.totalCentsEstimate;
+        }
+      }
+
+      return {
+        currentPrice,
+        workspaceId: workspace.id,
+      };
+    });
+
+    io.logger.info("list workspaces that exceeded their budgeted usage");
+    const exceededBudgetedAmount: {
+      budgetId: string;
+      emails: string[];
+      tenantId: string;
+      workspaceId: string;
+      workspaceName: string;
+      budgetedAmount: number;
+      currentPeriodBilling: number;
+    }[] = [];
+    workspaces.forEach((workspace) => {
+      const currentPrice = currentUsages.find(
+        (usage) => usage.workspaceId === workspace.id,
+      )?.currentPrice;
+
+      if (!currentPrice) {
+        return null;
+      }
+
+      const budget = workspace.budgets
+        .filter(isActiveBudget)
+        .reduce<(typeof workspace.budgets)[number] | null>((acc, budget) => {
+          // Check if current price exceeds the greatest budgeted amount
+          if (
+            currentPrice >= budget.fixedAmount &&
+            (!acc || budget.fixedAmount > acc.fixedAmount)
+          ) {
+            return budget;
+          }
+
+          return acc;
+        }, null);
+
+      if (budget) {
+        exceededBudgetedAmount.push({
+          budgetId: budget.id,
+          emails: budget.data.additionalEmails ?? [],
+          tenantId: workspace.tenantId,
+          workspaceId: workspace.id,
+          workspaceName: workspace.name,
+          budgetedAmount: budget.fixedAmount,
+          currentPeriodBilling: currentPrice,
+        });
+      }
+    });
+
+    // TODO: Assumming admin users will not change frequently their emails, we could store in in the budget entity to avoid this step.
+    await io.runTask(
+      "retrieve workpaces admin emails from Clerk for exceeded budgets",
```

Comment:
Since Unkey is not storing emails but retrieving them directly from Clerk, this job will need to retrieve the involved workspace admin's emails. So either i) retrieve it directly on the job or ii) save it on the budget and assume the admin user email won't change.